libchisel/src/remapimports.rs

```diff
@@ -0,0 +1,124 @@
+                (ImportPair::new("env", "ethereum_getCodeSize"), ImportPair::new("ethereum", "getCodeSize")),
+                (ImportPair::new("env", "ethereum_codeCopy"), ImportPair::new("ethereum", "codeCopy")),
+                (ImportPair::new("env", "ethereum_getReturnDataSize"), ImportPair::new("ethereum", "getReturnDataSize")),
+                (ImportPair::new("env", "ethereum_returnDataCopy"), ImportPair::new("ethereum", "returnDataCopy")),
+                (ImportPair::new("env", "ethereum_call"), ImportPair::new("ethereum", "call")),
+                (ImportPair::new("env", "ethereum_callCode"), ImportPair::new("ethereum", "callCode")),
+                (ImportPair::new("env", "ethereum_callDelegate"), ImportPair::new("ethereum", "callDelegate")),
+                (ImportPair::new("env", "ethereum_callStatic"), ImportPair::new("ethereum", "callStatic")),
+                (ImportPair::new("env", "ethereum_storageLoad"), ImportPair::new("ethereum", "storageLoad")),
+                (ImportPair::new("env", "ethereum_storageStore"), ImportPair::new("ethereum", "storageStore")),
+                (ImportPair::new("env", "ethereum_revert"), ImportPair::new("ethereum", "revert")),
+                (ImportPair::new("env", "ethereum_finish"), ImportPair::new("ethereum", "finish")),
+                (ImportPair::new("env", "ethereum_selfDestruct"), ImportPair::new("ethereum", "selfDestruct")),
+            ].iter().cloned().collect();
+        Translations { translations: trans }
+    }
+
+    fn insert(&mut self, from_module: &str, from_field: &str, to_module: &str, to_field: &str) {
+        self.translations.insert(ImportPair::new(from_module, from_field), ImportPair::new(to_module, to_field));
+    }
+
+//    fn get_simple(&self, module: &str, field: &str) -> Option<&str, &str> {
+//        if let Some(translation) = self.translations.get(&ImportPair::new(module, field)) {
+//            Some(translation.module.clone(), translation.field.clone())
+//        } else {
+//            None
+//        }
+//    }
+
+    fn get(&self, pair: &ImportPair) -> Option<&ImportPair> {
+        self.translations.get(&pair)
+    }
+}
+
+pub struct RemapImports {
+   translations: Translations
+}
+
+impl ModuleTranslator for RemapImports {
+    fn translate(self, module: &mut Module) -> Result<(), String> {
+        rename_imports(module, self.translations);
+        Ok(())
+    }
+}
+
+// FIXME: There is no `Module::import_section_mut()`
+fn import_section_mut(module: &mut Module) -> Option<&mut ImportSection> {
+    for section in module.sections_mut() {
+        if let &mut Section::Import(ref mut import_section) = section { return Some(import_section); }
+    }
+    None
+}
+
+pub fn rename_imports(module: &mut Module, translations: Translations) {
+    if let Some(section) = import_section_mut(module) {
+        for entry in section.entries_mut().iter_mut() {
```

Comment:
I think this can be simplified a bit into something like
`*section = section.entries_mut().iter_mut().map(|x| translations.get(x.module(), x.field()).collect()`
If a more functional style is preferred.